File: Solutions/0401-500/443. String Compression/Solution.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        read, write = 0, 0

        while read < len(chars):
            ch = chars[read]
            count = 0

            while read < len(chars) and chars[read] == ch:
                count += 1
                read += 1
            
            chars[write] = ch
            write += 1

            if count > 1:
                for digit in str(count):
                    chars[write] = digit
                    write += 1
            
        return write

File: Solutions/0401-500/443. String Compression/Solution.java
class Solution 
{
    public int compress(char[] chars) 
    {
        int read = 0, write = 0;

        while(read < chars.length)
        {
            char ch = chars[read];
            int count = 0;

            while(read < chars.length && chars[read] == ch)
            {
                count++;
                read++;
            }

            chars[write] = ch;
            write++;

            if(count > 1)
            {
                for(char digit : String.valueOf(count).toCharArray())
                {
                    chars[write] = digit;
                    write++;
                }
            }
        }
        return write;
    }    
}
